pkgs/build-support/php/hooks/default.nix

```diff
@@ -0,0 +1,21 @@
+{ makeSetupHook
+, php
+, jq
+, moreutils
+}:
+
+{
+  composerRepositoryHook = makeSetupHook
+    {
+      name = "composer-repository-hook.sh";
+      propagatedBuildInputs = [ php jq moreutils ];
```

Comment:
Shouldn't this be propagatedNativeBuildInputs? Also why do we propagate at all?

Reply:
No. This hook is meant to be used in `nativeBuildInputs`, so the [resulting offset](https://nixos.org/manual/nixpkgs/unstable/#ssec-stdenv-dependencies-reference) will end up as `depsBuildTarget` `(-1, 1)` (`buildInputs` `(0, 1)` + `nativeBuildInputs` `(-1, 0 )`). That is what we want – the host offset (first number in the pair, the platform on which the programs should run) will point to build platform of the currently built package.

Ideally, we would use `depsHostHostPropagated` so that `depsHostHost` `(0, 0)` + `nativeBuildInputs` `(-1, 0 )` = `nativeBuildInputs` `(-1, 0 )` but the target offset is irrelevant here so the difference does not really matter.

Propagation is needed because the setup hook is just a shell script expecting stuff on `PATH`. Other option would be using `substitutions` to hardcode program paths in the script, or using resholved.